### PR TITLE
Support XDG Base Directory Specification for `$SIDEBAR_DIR`

### LINKED ISSUE #40: Support XDG Base Directory Specification for `SIDEBAR_DIR`
Tmux support putting the config in the XDG path `$XDG_CONFIG_PATH/tmux/tmux.conf` and I  noticed that this plugin will still create a file `$HOME/.tmux/sidebar/directory_widths.txt`, thus polluting $HOME.

# Background
Lately the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) has been gaining quite a steady [popularity](https://wiki.archlinux.org/title/XDG_Base_Directory#Support), even though it's been around for a long time already. The basic idea is to standardize where programs put their files e.g. all config files in a certain directory. The benefits for end users are many, including:

* No polluted $HOME/
* Can easily grep for only config files in `$XDG_CONFIG_HOME`
* Can decide where to store configurations in more complex setups (multi system setup e.g.)
* Easily wipe state by removing `$XDG_STATE_HOME` or even `$XDG_DATA_HOME`, but without destroying and configuration files.

The first point is maybe what most end users enjoy the most though.

# How
Specifically for this project, this should mean that the search for the configuration file should be done like:

1. **if** `$XDG_CONFIG_HOME` is *set*, look for `$XDG_CONFIG_HOME/tmux/sidebar/directory_widths.txt`
1. **elseif** `$XDG_CONFIG_HOME` is *unset*, use the default value for the envvar which is `$HOME/.config`. Thus, the config file to look for is at `$HOME/.config/tmux/sidebar/directory_widths.txt`
1. **else** fall-back to current legacy `$HOME/.tmux/sidebar/directory_widths.txt`

The first two steps to can be codifed in bash like:
 `${XDG_CONFIG_HOME:-$HOME/.config}/amethyst/amethyst.yml`

I noticed that `variables.sh` set base dir path as an envvar `$SIDEBAR_DIR`:
https://github.com/tmux-plugins/tmux-sidebar/blob/aacbdb45bc5ab69db448a72de4155d0b8dbac677/scripts/variables.sh#L26

Maybe it's possible to override this value as a workaround until full XDG support is added? 🤔

### PR DESCRIPTION
**Pull request by erikw**
See #40 for background

Avoid cluttering `$HOME` by using the XDG dir as first priority (while keep using the old path if it exist to not break the workflow for existing users) for `$SIDEBAR_DIR`.

1. **If** `$XDG_CONFIG_HOME/tmux/sidebar` (or default `$HOME/.config/...`) exist, use this.
1. **elseif** `$HOME/.tmux/sidebar`exist, use this.
1. **else** default to XDG i.e. use `$XDG_CONFIG_HOME/tmux/sidebar` (or default `$HOME/.config/...`)

This PR worked on my computer™

Fixes #40

### CODE REVIEW

**Comment by bruno-** [comment]
The PR looks ok.
Should we use `$XDG_CONFIG_HOME ` or `$XDG_DATA_HOME` for this?

I'm leaning slightly towards `$XDG_DATA_HOME` because I've never manually edited this sidebar file. It's more of a sideproduct of using a plugin.

File: scripts/variables.sh [hunk 1]
```diff
@@ -23,4 +23,13 @@ TREE_WIDTH_OPTION="@sidebar-tree-width"
...
+SIDEBAR_DIR_XDG="${XDG_CONFIG_HOME:-$HOME/.config}/tmux/sidebar"
+SIDEBAR_DIR_OLD="$HOME/.tmux/sidebar"
+if [ -d "$SIDEBAR_DIR_XDG" ]; then
+	SIDEBAR_DIR="$SIDEBAR_DIR_XDG"
+elif [ -d "$SIDEBAR_DIR_OLD" ]; then
+	SIDEBAR_DIR="$SIDEBAR_DIR_OLD"
+else
+	SIDEBAR_DIR="$SIDEBAR_DIR_XDG"
+fi
+unset SIDEBAR_DIR_XDG SIDEBAR_DIR_DEFAULT
```
Comment (bruno-) [inline scripts/variables.sh:35]: What do you think about creating a small function `sidebar_dir` for this?
The variables inside can be local `local sidebar_dir_xdg=...`. Then we could have only one global variable:

`SIDEBAR_DIR="$(sidebar_dir)"`

Comment (erikw) [reply scripts/variables.sh]: ✅ makes sense!

**Comment by erikw** [comment]
> Should we use `$XDG_CONFIG_HOME ` or `$XDG_DATA_HOME` for this?

I like it, this is a better place. 

The PR is updated :)

**Comment by bruno-** [comment]
Thanks!

**Comment by eggbean** [comment]
I had forked this and used (more recent) `$XDG_STATE_HOME` which is really the ideal place for this.

**Pull request merged**